keras/layers/pooling.py

```diff
@@ -424,7 +444,21 @@ class GlobalAveragePooling1D(_GlobalPooling1D):
     '''
 
     def call(self, x, mask=None):
-        return K.mean(x, axis=1)
+        if mask is None:
+            return K.mean(x, axis=1)
+        else:
+            mask = K.cast(mask, K.floatx())
+            # Expand count so we can do a broadcast division
+            count = K.sum(mask, axis=1)
+            count = K.expand_dims(count, dim=1)
+            if K.backend() == 'tensorflow':  # tensorflow shape command dosn't work
```

Comment:
I don't know why just shape didn't work with tensorflow, I didn't get a helpful error message back when I wrote this code. I also had to cast the mask to a floatx because tensorflow would throw errors on it otherwise. There have been a number of threads discussions issue relating to backend code that would work with theano and not tensorflow, so I don't think this is a new issue. 

We also shouldn't have stuff like "int_shape" which is documented as part of the backend but only defined for the tensorflow backend. But it exists.